modules/csharp_api/csharp/preprocess/prepost_processor.cs

```diff
@@ -0,0 +1,178 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OpenVinoSharp.preprocess
+{
+
+    /// <summary>
+    /// Main class for adding pre- and post- processing steps to existing ov::Model
+    /// </summary>
+    /// <remarks>
+    /// This is a helper class for writing easy pre- and post- processing operations on ov::Model object assuming that
+    /// any preprocess operation takes one input and produces one output.
+    ///
+    /// For advanced preprocessing scenarios, like combining several functions with multiple inputs/outputs into one,
+    /// client's code can use transformation passes over ov::Model
+    /// </remarks>
+    public class PrePostProcessor
+    {
+        /// <summary>
+        /// [private]PrePostProcessor class pointer.
+        /// </summary>
+        public IntPtr m_ptr = IntPtr.Zero;
+
+        /// <summary>
+        /// [public]PrePostProcessor class pointer.
+        /// </summary>
+        public IntPtr Ptr { get { return m_ptr; } set { m_ptr = value; } }
+
+        /// <summary>
+        /// Default construction through Model.
+        /// </summary>
+        /// <param name="model">model.</param>
+        public PrePostProcessor(Model model)
+        {
+            ExceptionStatus status = (ExceptionStatus)NativeMethods.ov_preprocess_prepostprocessor_create(model.Ptr, ref m_ptr);
+            if (status != 0)
+            {
+                System.Diagnostics.Debug.WriteLine("PrePostProcessor init error : {0}!", status.ToString());
+            }
+        }
+        /// <summary>
+        /// Default destructor
+        /// </summary>
+        ~PrePostProcessor() { dispose(); }
+        /// <summary>
+        /// Release unmanaged resources
+        /// </summary>
+        public void dispose() {
```

Comment:
In C# you should use `Dispose`, otherwise nothing will work with `using` statement

Reply:
Also, please just implement IDispose interface to have all benefits `try-with-resources` in C# World